### ISSUE TITLE
slaveOk=false - hot to overcome?

### ISSUE THREAD

**Issue by PavelPolyakov** [hapijs/catbox-mongodb#12]
Hi,

Currently I'm trying to configure the catbox-mongodb cache to interact with my mongodb cluster.

Unfortunately, I have the next error:
![image](https://cloud.githubusercontent.com/assets/839290/6670615/563d0eee-cc08-11e4-9396-3b77b3330e08.png)

Could anyone say to me how could I overcome it?

Regards,

**Comment by jardakotesovec** [comment]
Hi Pavel,
I have not used mongodb cluster, so you need to give me clue. I guess you need some extra parameters for connection here?

https://github.com/hapijs/catbox-mongodb/blob/master/lib/index.js#L78

**Comment by PavelPolyakov** [comment]
@jardakotesovec 

I believe so, thanks for quick answer.

It looks that the ```slaveOk``` configuration option could solve the issue:
http://mongodb.github.io/node-mongodb-native/1.4/api-generated/db.html

**Comment by jardakotesovec** [comment]
Lets first understand what exactly you want to achieve before changing code.

You created [replica set](http://docs.mongodb.org/manual/core/replication-introduction/), right? Have you tried to connect directly using just native mongodb driver?

**Comment by PavelPolyakov** [comment]
@jardakotesovec 
I'm sorry that had bothered you. It turned out that I used the wrong port.

When I stated host:27017 - I had the error, but it turned out that I should use the 27019 instead. So the code is working just fine.

One small issue - currently, my mongo url for the native driver looks the next way:
~~~
mongodb://username:password@host1:27019,host2:27019,host3:27019/database
~~~

As you see - 3 hosts are stated, as I understand mongo driver would use this information and spread the pressure on the hosts, or, would be able to connect the 2nd host when the 1st is down.

In the catbox-mongodb, we use another kind of connection notation, and, as I understand, we're able to point just one host there, am I right? Or we can solve this as well.

Thanks for the help.

Regards,

**Comment by jardakotesovec** [comment]
You are right that catbox-mongodb is using old api and is possible to set up only one host.  

It would makes sense to update it to newer MongoClient api - http://mongodb.github.io/node-mongodb-native/driver-articles/mongoclient.html . Ideally without breaking changes.

Not sure when I get into it. Would you consider pull request?

**Comment by jardakotesovec** [comment]
Implemented in #20
